theano/sandbox/gpuarray/opt.py

```diff
@@ -260,6 +261,25 @@ def local_gpu_elemwise(node):
     return res
 
 
+@register_opt()
+@gof.local_optimizer([GpuElemwise])
+def cast_elemwise_pow_exponent(node):
```

Comment:
I added tests. Now, if the base is float32 or float64, the exponent can have any uint, int or float dtype and it works. However, there are still cases where it doesn't work. If both the base and the exponent are ints, for instance, the output dtype will also be int and it will not fit in the two patterns that the gpu implementation supports at this point in time (float32,float32)->(float32) and (float64,float64)->(float64).

I am not sure which way would be best to handle this.
We could add casts before the inputs and after the output when moving the Elemwise(pow) to the Gpu but that means in some cases we would have to downcast the output (say from float32 to int8). 
We could also not move the Elemwise(pow) nodes with unsupported dtypes to gpu but if the goal of the new backend was to support other dtypes than floating points, then that's probably not the best solution.
Any opinion on this?